### DISCUSSION TITLE
GSI without a sort key

### DISCUSSION THREAD

**Discussion by mathieubrd** [typedorm/typedorm#297] [Q&A]
Why it is not possible to have a GSI without any sort key?
For example, my use case is to retrieve all items of the same entity (e.g. retrieve all users).
So I would create my entity like so:

```(typescript)
@Entity({
  name: "User",
  primaryKey: {
    partitionKey: "USER#{{email}}",
    sortKey: "USER#{{email}}",
  },
  indexes: {
    entity: {
      type: INDEX_TYPE.GSI,
      partitionKey: "User",
      // No need to have a sort key for this GSI
    },
  },
})
```

To retrieve all my users, I can query on my `entity` GSI with `entity = User`
But `Index` is typed in a way I cannot provide only a partition key.
I try to put an empty string in the sort key but nothing is inserted in my GSI.

Or maybe there is another way to fullfill my use case?

**Comment by dash-tobin** [comment]
This doesn't answer your question, but provides a solution. You can use empty string.

```
@Entity({
  name: "User",
  primaryKey: {
    partitionKey: "USER#{{email}}",
    sortKey: "USER#{{email}}",
  },
  indexes: {
    entity: {
      type: INDEX_TYPE.GSI,
      partitionKey: "User",
      sortKey: '',
    },
  },
})

```

**Comment by whimzyLive** [comment]
@mathieubrd @dash-tobin 
Apologies, this got missed from the refinement.

Yes, It is completely valid to have GSIs without `sortKey`, and therefore it should be allowed. 

I'll pull this in as an enhancement, to be added in 1.16.x

**Comment by whimzyLive** [comment]
tracked as #335
